fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java

```diff
@@ -1662,10 +1664,10 @@ public void modifyApplication(final JsonCommand command, final Map<String, Objec
             this.accountNumber = StringUtils.defaultIfEmpty(newValue, null);
         }
 
-        if (command.isChangeInStringParameterNamed(SavingsApiConstants.externalIdParamName, this.externalId)) {
+        if (command.isChangeInStringParameterNamed(SavingsApiConstants.externalIdParamName, this.externalId.getValue())) {
             final String newValue = command.stringValueOfParameterNamed(SavingsApiConstants.externalIdParamName);
             actualChanges.put(SavingsApiConstants.externalIdParamName, newValue);
-            this.externalId = StringUtils.defaultIfEmpty(newValue, null);
+            this.externalId = ExternalIdFactory.produce(newValue);
```

Comment:
Here we should generate one if nothing was provided: `ExternalIdFactory.create(...)`

Reply:
Since it is an Entity class, and the create method is not static, I should pass the `ExternalIdFactory` dependency as a method parameter, etc. So it won't be very pretty I think. Is it a business requirement? In the other cases, I'll implement your change request, only this case I'm a bit hesitant.

Reply:
Till this logic is not extracted out from the entity you might wanna use this:
```
            ExternalId externalId = ExternalIdFactory.produce(newValue);
            if (externalId.isEmpty() && TemporaryConfigurationServiceContainer.isExternalIdAutoGenerationEnabled()) {
                externalId = ExternalId.generate();
            }
```

Reply:
Thanks! Updated the PR according to this.